### PR TITLE
Clarify readme

### PR DESCRIPTION
**Pull request by ncopa**
**What this PR Includes**

Try make it slightly clearer what k0s is and does, and how it is different from other kubernetes distros by
    
- Remove redundant and non essential text
- List key features early (above the demo)
- Move build and smoke tests to the bottom

### CODE REVIEW

**Comment by ncopa** [comment]
I think I forgot to add a line about where to find the `/var/lib/k0s/pki/admin.conf`

File: README.md [hunk 1]
```diff
@@ -75,38 +77,6 @@ While some Kubernetes distros package everything and the kitchen sink in, k0s tr
...
-## Smoke test
-
-To run a smoke test after build:
-```
-make check-basic
-```
-
 ## Cluster bootstrapping
 
 Move the built `k0s` binary to each of the nodes.
```
Comment (jasmingacic) [inline README.md:82]: I'd suggest adding this too:
```
curl -SsLf get.k0s.sh 
```

File: README.md [hunk 2]
```diff
@@ -128,6 +98,8 @@ Join a new worker node to the cluster by running:
 k0s worker "superlongtokenfrompreviousphase"
 ```
 
+The client config (for kubectl or lens) can be found in `/var/lib/k0s/pki/admin.conf` on the controller.
+
```
Comment (jasmingacic) [inline README.md:102]: We should also point out that "localhost" needs to be replaced if hosted elsewhere than cloud.

**Review by jasmingacic — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jasmingacic) [reply README.md:82]: I can't comment on the line below but I'd also give an example with `enable-worker`

Comment (ncopa) [reply README.md:82]: Do we need `--enable-worker` example in the readme when we have it in the [FAQ](https://github.com/k0sproject/k0s/blob/main/docs/FAQ.md#how-do-i-run-a-single-node-cluster)?

Comment (jasmingacic) [reply README.md:82]: I'm thinking it is most likely how someone is going to try k0s

**Review by jasmingacic — APPROVED** [review]
looking good, let's ship it

**Review by jnummelin — APPROVED** [review]

**Pull request merged**